internal/components/multi_endpoint.go

```diff
@@ -134,10 +135,16 @@ func (mp MultiPortBuilder[ComponentConfigType]) Build() (*MultiPortReceiver, err
 		return nil, fmt.Errorf("must provide at least one port mapping")
 	}
 
+	defaultRecAddr := ""
+	if mp[0].MustBuild().settings != nil {
```

Comment:
no need for the nil check. also we may as well just save the built thing and use it here and below for the name